### DISCUSSION TITLE
pac auth create - difference in behaviour between dotnet tool pac and VSCode extension pac

### DISCUSSION THREAD

**Discussion by rajyraman** [microsoft/powerplatform-build-tools#416] [Ideas]
I ran the same command `pac auth create -env ENVURL` in Codespaces one with Power Platform Tools extension installed and one as just dotnet tool. The Power Platform Tools extension pac does this

<img width="457" alt="image" src="https://github.com/microsoft/powerplatform-build-tools/assets/5035266/41edc5d6-09ff-428f-83c3-9c4b7a598f7e">

dotnet tool on the other hand opens a popup and redirects to localhost after login. But this fails because pac is running inside Codespaces.

Please make dotnet tool pac do the same thing that the Power Platform Tools pac is doing.

**Comment by petrochuk** [comment]
Moving to discussions

**Reply by petrochuk** [reply]
Have you tried [--deviceCode](https://learn.microsoft.com/en-us/power-platform/developer/cli/reference/auth#--devicecode--dc)?

```
pac auth create --deviceCode
